projects/rero/ng-core/src/lib/record/record-ui.service.ts

```diff
@@ -172,4 +175,32 @@ export class RecordUiService {
 
     return of({ can: true, message: '' });
   }
+
+  /**
+   * Redirects after saving a resource
+   * @param pid - string, pid of the record
+   * @param record - object, record to save
+   * @param recordType - string, Type of resource
+   * @param action - string, http action: create or update
+   */
+  redirectAfterSave(pid: string, record: any, recordType: string, action: string) {
+    const config = this.getResourceConfig(recordType);
+    if (config.redirectUrl) {
+      config.redirectUrl(record).subscribe((result: string) => {
+        if (result !== null) {
+          this.router.navigate([result]);
+        }
+      });
+    } else {
+      // Default behaviour
+      if (action === 'update') {
+        // Relative path not working
+        // this.router.navigate(['../../detail', record.pid], {relativeTo: this.route, replaceUrl: true});
```

Comment:
Why this is not working? It worked before...

Reply:
It doesn't work when used in a service. We have the same case in rero-ils-ui: https://github.com/rero/rero-ils-ui/blob/bcaf8410dee9d4224a6cde48c78f69eac2b3c58b/projects/admin/src/app/record/custom-editor/circulation-settings/circulation-policy.service.ts#L82